Packs/MicrosoftExchangeOnline/Integrations/EwsExtensionEXOPowershellV3/EwsExtensionEXOPowershellV3.ps1

```diff
@@ -0,0 +1,1572 @@
+        $cmd_params = @{ }
+        $entries_array = @()
+        $ids_array = @()
+        if ($entries.Contains(','))
+        {
+            $entries_array = $entries -split ','
+        }
+        if ($entries -and -not $entries_array)
+        {
+            $entries_array += $entries
+            $cmd_params.Entries = $entries_array
+        }
+        elseif ($entries_array)
+        {
+            $cmd_params.Entries = $entries_array
+        }
+        if ($ids -and $ids.Contains(','))
+        {
+            $ids_array = $ids -split ','
+        }
+        if ($ids -and -not $ids_array)
+        {
+            $ids_array += $ids
+            $cmd_params.Ids = $ids_array
+        }
+        elseif ($ids_array)
+        {
+            $cmd_params.Ids = $ids_array
+        }
+        if ($list_type)
+        {
+            $cmd_params.ListType = $list_type
+        }
+        if ($list_subtype)
+        {
+            $cmd_params.ListSubType = $list_subtype
+        }
+        $this.CreateSession()
+        $results = Remove-TenantAllowBlockListItems @cmd_params
+        $this.DisconnectSession()
+        return $results
+        <#
+            .DESCRIPTION
+            Use the New-TenantAllowBlockListItems cmdlet to add new entries to the Tenant Allow/Block Lists for your organization.
+            This cmdlet returns all new entries created, details about them, and their status
+
+        #>
+    }
+
+    [PSObject]
+    EXOGetTenantAllowBlockList(
+            [string]$entry,
+            [string]$list_type,
+            [string]$list_subtype,
+            [string]$action,
+            [bool]$no_expiration,
+            [string]$expiration_date
+    )
+    {
+        $cmd_params = @{ }
+        if ($entry)
+        {
+            $cmd_params.Entry = $entry
+        }
+        if ($list_type)
+        {
+            $cmd_params.ListType = $list_type
+        }
+        if ($list_subtype)
+        {
+            $cmd_params.ListSubType = $list_subtype
+        }
+        if ($no_expiration)
+        {
+            $cmd_params.NoExpiration = $null
+        }
+        if ($expiration_date)
+        {
+            $cmd_params.ExpirationDate = $expiration_date
+        }
+        if ($action -eq "Block")
+        {
+            $cmd_params.Block = $null
+        }
+        if ($action -eq "Allow")
+        {
+            $cmd_params.Allow = $null
+        }
+        $this.CreateSession()
+        $results = Get-TenantAllowBlockListItems @cmd_params
+        $this.DisconnectSession()
+        return $results
+        <#
+            .DESCRIPTION
+            Use the Get-TenantAllowBlockListItems cmdlet to retrieve current entries in the Tenant Allow/Block Lists for your organization.
+            This cmdlet returns current entries and details about them.
+
+        #>
+    }
+    [psobject]GetJunkRules([string]$mailbox) {
+        $response = ''
+        try {
+            # Establish session to remote
+            $this.CreateSession()
```

Comment:
Why do some commands have `CreateSession()` in `try block` and `DisconnectSession()` in `finally block` ( which is probably best practice)
And some dont ?

Reply:
Full disclosure, this was smushing together v1 and v2 so there are some minor differences. But for best practices, I'll add these here too.